qcodes/instrument/visa.py

```diff
@@ -119,6 +128,7 @@ def __init__(
 
         self.set_terminator(terminator)
         self.timeout.set(timeout)
+        finalize(self, _close_visa_handle, self.visa_handle, str(self.name))
```

Comment:
shall we move this line right below self.visa_handle is assigned? so that possible exceptions in the rest of the init can't prevent the weakref.finalize to be "armed"?